### DISCUSSION TITLE
GeoPackageFragment load gpkg file

### DISCUSSION THREAD

**Discussion by CodeK1988** [WorldWindEarth/WorldWindKotlin#51] [General]
hi mate,   load gpkg file  get error message "java.lang.IllegalArgumentException: Angle.init: NaN is not supported!"
`
val geoPackageFile = TutorialUtil.unpackAsset(requireContext(), "rivers.gpkg")
val layer = GpkgLayerFactory.createLayer(geoPackageFile.path)
`

**Comment by EMaksymenko** [comment]
This issue is related to your file.

Please debug why angle is missed on your specific case, or attach rivers.gpkg file to this topic.

**Comment by CodeK1988** [comment]
[rivers.zip](https://github.com/WorldWindEarth/WorldWindKotlin/files/13437299/rivers.zip)

**Reply by EMaksymenko** [reply]
This file contains one raster tiles layer named river_tiles, but its matrix set is defined in meters.
WorldWind Kotlin currently suports only matrix sets defined in degrees.

We do not have resources now to add metric time matrix support. If you wish, you may contribute.
<img width="599" alt="image" src="https://github.com/WorldWindEarth/WorldWindKotlin/assets/12487306/24213cd1-0cc1-4b59-a9dd-5a12f2640d81">

**Reply by CodeK1988** [reply]
now get error "(1) no such table: gpkg_2d_gridded_coverage_ancillary"

**Reply by CodeK1988** [reply]
@ComBatVision Hi mate, how can I convert mbtiles in pbf format into png mbtiles files? Ultimately I want to convert png format mbtiles map files to gpkg.

**Reply by EMaksymenko** [reply]
I do not know what is pbf format. Please, search in the internet.
